Xamarin.Forms.Platform.iOS/Renderers/LabelRenderer.cs

```diff
@@ -469,5 +485,46 @@ void UpdateMaxLines()
 #endif
 			}
 		}
+
+		void UpdatePadding()
+		{
+			if (Element.Padding.IsEmpty)
+				return;
+
+#if __MOBILE__
+			var formsLabel = Control as FormsLabel;
+			if (formsLabel == null)
+			{
+				Debug.WriteLine($"{nameof(LabelRenderer)}: On iOS, a Label created with no padding will ignore padding changes");
+				return;
+			}
+
+			formsLabel.TextInsets = new UIEdgeInsets(
+					(float)Element.Padding.Top,
+					(float)Element.Padding.Left,
+					(float)Element.Padding.Bottom,
+					(float)Element.Padding.Right);
+			UpdateLayout();
+#endif
+		}
+
+#if __MOBILE__
+		class FormsLabel : NativeLabel
+		{
+			public UIEdgeInsets TextInsets { get; set; }
+
+			public FormsLabel(RectangleF frame) : base(frame)
+			{
+			}
+
+			public override void DrawText(RectangleF rect) => base.DrawText(TextInsets.InsetRect(rect));
+
+			public override SizeF SizeThatFits(SizeF size) => AddInsets(base.SizeThatFits(size));
+
+			SizeF AddInsets(SizeF size) => new SizeF(
+				width: size.Width + TextInsets.Left + TextInsets.Right,
+				height: size.Height + TextInsets.Top + TextInsets.Bottom);
+		}
```

Comment:
Have you considered overriding `intrinsicContentSize` as well?  See https://stackoverflow.com/a/27461090/246392

Reply:
intrinsicContentSize is for auto layout AFAIK.